include/sound/sof/topology.h

```diff
@@ -39,6 +39,12 @@ enum sof_comp_type {
 	SOF_COMP_ASRC,		/**< Asynchronous sample rate converter */
 	SOF_COMP_DCBLOCK,
 	SOF_COMP_SMART_AMP,             /**< smart amplifier component */
+
+	/*
+	 * No more _COMP_ types to be added.
+	 * Use SOF_COMP_PROCESS from now on.
```

Comment:
@keyonjie @lgirdwood  Shouldn't this be "No more _COMP_ types should be added for effects" (snd_soc_dapm_effect/sof_ipc_comp_process)?

If want to have main type that defines the ALSa controls and IPC msg, and flavor which can vary, we can't put everything under same component id right? The above comment would mean our current list of main types is exhaustive and it will never be extended. This doesn't make sense. If we add a new component class, it should be identified by a unique component type.

Reply:
Yes, let me change it to "No more _COMP_ types should be added for effects/process".